.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,64 @@
+name: integration-tests
+on:
+  push:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+  pull_request:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+
+env:
+  STORAGE_PATH: "/tmp"
+
+jobs:
+  deploy-and-test:
+    strategy:
+      matrix:
+        key: [8, 16]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checks-out repo
+        uses: actions/checkout@v2
+      - name: Install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install deps
+        run: |
+          /usr/bin/pip3 install -r ./requirements.txt
+        working-directory: ./integration-tests
+      - name: Install compose
+        uses: KengoTODA/actions-setup-docker-compose@main
+        with:
+          version: '2.6.0'
+      - name: Get version data
+        run: |
+          echo "VERSION=$(cat bob-apps/Cargo.toml | grep ^version | head -1 | awk '{print $3}' | sed -e 's/"//g' | sed -e 's/-/./g')" >> $GITHUB_ENV
+          echo "VERSION_NO_FORMATTING=$(cat bob-apps/Cargo.toml | grep ^version | head -1 | awk '{print $3}' | sed -e 's/"//g')" >> $GITHUB_ENV
+      - name: Get tools
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: build-packages.yml
```

Comment:
You should build binaries as a separate job of this workflow, because there is no guarantee, that `build-packages.yml` will be run every time this workflow is run.

Additionally, I think it is better to build docker image, not the binaries

Reply:
I see, that you use `ccg` and `bobp` binaries, so you should have 2 additional jobs: 
1. Build binaries
2. Build docker image